docs/DESKTOP_ENVIRONMENTS.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Desktop Environment integration
+category: Concepts
+layout: default
+---
+
+# Desktop Environments
+
+NOTE: This document is work-in-progress.
+
+## Single Graphical Session
+
+To be able to manage a graphical session using systemd, there is a strong
+requirement that each user only has one such session running at a time. While
+this might not have always been the case historically, having multiple sessions
+for one user running at the same time is problematic for a number of reasons
+(including e.g. the DBus session bus being shared).
+
+Note that in principle it is possible to run a single graphical session
+across multiple logind seats in principle. i.e. when the user logs in to a
+second seat, one could assign the seats resources to the existing session,
+effectively presenting it as a single seat.
+
+## Pre-defined systemd units
+
+[`systemd.special(7)`](https://www.freedesktop.org/software/systemd/man/systemd.special.html)
+defines the `graphical-session.target` and `graphical-session-pre.target` to
+allow cross-desktop integration. Furthermore, systemd defines the three base
+slices `background`, `apps` and `session`. All units should be placed into
+one of these slices depending on their purposes:
+
+ * `session.slice`: Contains only processes essential to run the users graphical session
+ * `apps.slice`: Contains all normal applications that the user is running
+ * `background.slice`: Useful for low-priority background tasks
+
+The purpose of this grouping is to assign different priorities to the
+applications. This could e.g. mean reserving memory to session processes,
+preferentially killing background tasks in out-of-memory situations
+or assinging different memory/CPU/IO priorities to ensure that the session
+runs smoothly under load.
+
+TODO: Will there be a default to place units into e.g. `apps.slice` by default
+rather than the root slice?
+
+## XDG standardization for applications
+
+To ensure cross-desktop compatibility and encourage sharing of good practices,
+desktop environments should adhere to the following conventions:
+
+ * Application units should follow the scheme `apps-launcher-ApplicationID-PID.service`,
+   e.g. `apps-gnome-org.gnome.Evince-12345.service`,
+   `apps-flatpak-org.telegram.desktop-12345.service` or `apps-KDE-org.kde.okular-12345.service`.
+ * Use of `.scope` units is discouraged.
+ * Applications must be placed into the appropriate slice. It is suggested to use
+   the same naming scheme, and use the `apps-launcher-ApplicationID.slice`
```

Comment:
Hmm, why? I don't think there's any particular reason to recommend that in general. If I have two instances of the same application, quite often they should be treated as completely separate as far as resource allocation goes.

Reply:
this is a bit misleading since a service is already a cgroup "domain" on which ressource controll can be applied. there is no need to add an extra layer of slices...

Reply:
I think it would be useful to check for existing slices with this naming scheme, that way optional persistent settings for an application can be achieved.

Reply:
I'll drop this again. It is perfectly acceptable to just have all applications be siblings off each other.

Note that systemd kind of does something similar for template units, implicitly creating a common slice for them default. Not sure if that is an argument either way though.